node/node.go

```diff
@@ -0,0 +1,452 @@
+	if err != nil {
+		return nil, nil, err
+	}
+	consensusDir := filepath.Join(dir, "consensus")
+	if err := os.MkdirAll(consensusDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	cm, errCh := mconsensus.New(g, cfg.Bootstrap, consensusDir)
+	select {
+	case err := <-errCh:
+		if err != nil {
+			return nil, nil, err
+		}
+	default:
+		go func() {
+			if err := <-errCh; err != nil {
+				log.Println("WARNING: consensus initialization returned an error:", err)
+			}
+		}()
+	}
+	tpoolDir := filepath.Join(dir, "transactionpool")
+	if err := os.MkdirAll(tpoolDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	tp, err := transactionpool.New(cm, g, tpoolDir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	walletDir := filepath.Join(dir, "wallet")
+	if err := os.MkdirAll(walletDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	walletAddr := wallet.StandardAddress(walletKey.PublicKey())
+	ws, ccid, err := stores.NewJSONWalletStore(walletDir, walletAddr)
+	if err != nil {
+		return nil, nil, err
+	} else if err := cm.ConsensusSetSubscribe(ws, ccid, nil); err != nil {
+		return nil, nil, err
+	}
+	w := wallet.NewSingleAddressWallet(walletKey, ws)
+
+	dbDir := filepath.Join(dir, "db")
+	if err := os.MkdirAll(dbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	dbConn := stores.NewSQLiteConnection(filepath.Join(dbDir, "db.sqlite"))
+
+	hostdbDir := filepath.Join(dir, "hostdb")
+	if err := os.MkdirAll(hostdbDir, 0700); err != nil {
+		return nil, nil, err
+	}
+	sqlStore, ccid, err := stores.NewSQLStore(dbConn, true)
+	if err != nil {
+		return nil, nil, err
+	} else if err := cm.ConsensusSetSubscribe(sqlStore, ccid, nil); err != nil {
+		return nil, nil, err
+	}
+
+	contractsDir := filepath.Join(dir, "contracts")
+	if err := os.MkdirAll(contractsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	objectsDir := filepath.Join(dir, "objects")
+	if err := os.MkdirAll(objectsDir, 0700); err != nil {
+		return nil, nil, err
+	}
+
+	cleanup := func() error {
+		errs := []error{
+			g.Close(),
+			cm.Close(),
+			tp.Close(),
+			sqlStore.Close(),
+		}
+		for _, err := range errs {
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	b := bus.New(syncer{g, tp}, chainManager{cm}, txpool{tp}, w, sqlStore, sqlStore, sqlStore, sqlStore)
+	return b, cleanup, nil
+}
+
+func newWorker(cfg WorkerConfig, walletKey consensus.PrivateKey) (*worker.Worker, func() error, error) {
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	workerKey := blake2b.Sum256(append([]byte("worker"), walletKey...))
+	w := worker.New(workerKey, b)
+	return w, func() error { return nil }, nil
+}
+
+func newAutopilot(cfg AutopilotConfig, dir string) (*autopilot.Autopilot, func() error, error) {
+	store, err := stores.NewJSONAutopilotStore(dir)
+	if err != nil {
+		return nil, nil, err
+	}
+	b := bus.NewClient(cfg.BusAddr, cfg.BusPassword)
+	w := worker.NewClient(cfg.WorkerAddr, cfg.WorkerPassword)
+	a, err := autopilot.New(store, b, w, cfg.LoopInterval)
```

Comment:
I think eventually an autopilot should be constructed with only a `bus` alongside a method that allows adding/removing workers from a worker pool

Reply:
Perhaps the autopilot's `Worker` interface could be *implemented* by a worker pool type? 🤔 
I suppose it depends on who should be responsible for managing the worker pool. Is it the autopilot's job to distribute work evenly among workers? Or should the autopilot just call `w.DoAThing`, and let `w` figure out which worker to use? I'm leaning towards the latter.

Reply:
Going back to our original definition of the autopilot I'd say we don't want the autopilot to distribute work. Work should be distributed even if the autopilot dies. Probably through some load balancer in front of the workers. The workers might have some endpoint to expose their load to make it easier for the load balancer. People could also just write their own load balancer depending on the information that the workers expose individually.

The only gain I see from having multiple workers in the autopilot is that it can still form/renew contracts and refill EAs even if a worker dies. 

Anyway that's probably a better discussion for Discord than this PR.

Reply:
Well user generated traffic would indeed be distributed among workers by some load balancer sitting in front of it, but I figured the autopilot would be able to do things like: check what worker is currently idle and have him form new contracts. Thinking about it more though I would now agree with Luke as I'm struggling to find a really good and compelling reason for the autopilot to manage the worker pool, or at least have full access to it.